src/main/java/Scheduler.java

```diff
@@ -22,43 +29,124 @@ public Scheduler(ArrayList<Course> courses, ArrayList<Schedule> schedules) {
         this.schedules = schedules;
     }
 
+    public void addFilters(List<Filter> filters) {
+        this.filters.addAll(filters);
+    }
+
     /**
-     * Takes a list of course codes and outputs a schedule that takes the first lecture session and
-     * first tutorial session in each lecture.
+     * Scheduler used to create courses all course permutations recursively. During generation, schedulers not passing
+     * filters are removed in the populatePermutations and extendPermutations().
      *
-     * @param courseCodes an ArrayList of course codes from which a schedule will be generated.
+     * @param newCourses courses sorted by priority to be scheduled.
      */
-    public Schedule createBasicSchedule(ArrayList<String> courseCodes) {
+    public List<Schedule> permutationScheduler(List<Course> newCourses) {
```

Comment:
so in general we shouldn't be doing this as a lump of all of the schedules, rather sorting and subdividing
i.e. F vs S but idk if you want to do that now or later

Reply:
later. we need to have a functioning scheduler to start